lib/deploy.js

```diff
@@ -434,6 +434,49 @@ function deployScheduleTrigger(name, namespace, schedule, options) {
     });
 }
 
+function deployHttpTrigger(name, namespace, path, options) {
+  if (options.ingress.enableTlsAcme && options.ingress.tlsSecretName) {
+    throw new Error('Cannot specify both enableTlsAcme and tlsSecretName');
+  }
+  // fix for issue https://github.com/kubeless/kubeless/issues/1124
+  const baseAnnotations = {
+    'kubernetes.io/ingress.class': options.ingress.class || 'nginx',
+  };
+  // end fix
+  const trigger = {
+    apiVersion: 'kubeless.io/v1beta1',
+    kind: 'HTTPTrigger',
+    metadata: {
+      name,
+      namespace,
+      labels: {
+        'created-by': 'kubeless',
+      },
+      annotations: Object.assign(baseAnnotations, options.ingress.additionalAnnotations),
+    },
+    spec: {
+      'basic-auth-secret': options.ingress.basicAuthSecretName || '',
+      'cors-enable': options.ingress.cors || false,
+      'function-name': name,
+      gateway: options.ingress.class || 'nginx',  // not working: https://github.com/kubeless/kubeless/issues/1124
```

Comment:
the comment is incorrect, right? it now works with these changes?

Reply:
Unfortunately no. The gateway option is ignored by the kubeless controller, as reported in this issue: https://github.com/kubeless/kubeless/issues/1124
I added the gateway option so when the fix will be relased it will works as expected but until then, I added a workaround that adds the ingress class annotation.

Reply:
ah, right, that was an issue in the controller